File: 1446. Consecutive Characters/slidingwindow.cpp
class Solution {
public:
    int maxPower(string s) {
        if(s.size() == 0)
            return 0;
        int len = s.size(), ret = 1, last = 0, index = 0;
        while(index < len){
            char c = s[index];
            while(index < len && s[index] == c){
                ret = max(ret, index - last + 1);
                index++;
            }
            last = index;
        }
        return ret;
    }
};

File: offer/JZ48 最长不含重复字符的子字符串/slidingwindow.java
import java.util.*;

public class Solution {
    public int lengthOfLongestSubstring (String s) {
        if(s.length() == 0)
            return 0;
        Map<Character, Integer> map = new HashMap<>();
        map.put(s.charAt(0), 0);
        int len = s.length(), ret = 1, start = 0;
        
        for(int i = 1; i < len; ++i){
            char c = s.charAt(i);
            if(map.containsKey(c)){
                start = map.get(c) + 1;
                removeExpireChar(map, start);
            }
            map.put(c, i);
            ret = Math.max(ret, i - start + 1);
        }
        return ret;
    }
    
    private void removeExpireChar(Map<Character, Integer> map, int index){
        Iterator<Map.Entry<Character, Integer>> iterator = map.entrySet().iterator();
        while (iterator.hasNext()) {
            Map.Entry<Character, Integer> it = iterator.next();
            if(it.getValue() < index)
                iterator.remove();
        }
    }
}